avocado/plugins/runners/exec_test.py

```diff
@@ -185,6 +167,28 @@ def run(self, runnable):
             ):
                 most_current_execution_state_time = now
                 yield self.prepare_status("running")
+            time.sleep(RUNNER_RUN_CHECK_INTERVAL)
+
+    def run(self, runnable):
+        try:
+            process = self._run_proc(runnable)
+        except Exception as e:
+            yield self.prepare_status("started")
+            yield self.prepare_status(
+                "finished", {"result": "error", "fail_reason": str(e)}
+            )
+            self._cleanup(runnable)
+            return
+
+        def poll_proc():
+            return process.poll() is not None
+
+        yield self.prepare_status("started")
```

Comment:
I know that this is not totally related to the change, but I think it could be handled here as well. The test execution actually starting on line 174, IMO we should move `started` message before that. With this change, we can also remove line 176.

Reply:
Yeah, that makes sense... bonus is we'd avoid having a duplicate "started" message on the exception handler.  I'm putting this diff on a new commit:

```diff
diff --git a/avocado/plugins/runners/exec_test.py b/avocado/plugins/runners/exec_test.py
index 0adfcd6e1..a72fd8644 100644
--- a/avocado/plugins/runners/exec_test.py
+++ b/avocado/plugins/runners/exec_test.py
@@ -154,17 +154,17 @@ class ExecTestRunner(BaseRunner):
         )
 
     def run(self, runnable):
+        yield self.prepare_status("started")
+
         try:
             process = self._run_proc(runnable)
         except Exception as e:
-            yield self.prepare_status("started")
             yield self.prepare_status(
                 "finished", {"result": "error", "fail_reason": str(e)}
             )
             self._cleanup(runnable)
             return
 
-        yield self.prepare_status("started")
         most_current_execution_state_time = None
         timeout = RUNNER_RUN_CHECK_INTERVAL
         while process.returncode is None:
```

Let me know if it looks OK to you.

Reply:
yes, this is exactly what I was proposing, thanks.